[SymmMem] Move code to where it is used

kwen2501 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #157611
* #157513

`maybe_initialize_env_vars` and `initialize_nvshmem_with_store` are only used in `NVSHMEMSymmetricMemory.cu`. Moving them there.

cc @H-Huang @awgu @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k

kwen2501 committed:
Update

[ghstack-poisoned]

kwen2501 committed:
Update

[ghstack-poisoned]

kwen2501 committed:
Update

[ghstack-poisoned]

Skylion007 commented on torch/csrc/distributed/c10d/symm_mem/NVSHMEMSymmetricMemory.cu:
```diff
@@ -263,6 +263,64 @@ class NVSHMEMSymmetricMemory : public SymmetricMemory {
+  // Print version
+  int major, minor;
+  ::nvshmem_info_get_version(&major, &minor);
+  LOG(INFO) << "NVSHMEM is available, version: " << major << "." << minor;
```
```suggestion
  LOG(INFO) << "NVSHMEM is available, version: " << major << '.' << minor;

Skylion007 commented on torch/csrc/distributed/c10d/symm_mem/nvshmem_extension.cuh:
```diff
@@ -1,16 +1,19 @@
+    int result = (stmt);                                                     \
+    TORCH_CHECK(                                                             \
+        result == 0,                                                         \
+        std::string(__FILE__) + ":" + std::to_string(__LINE__) + " " + msg + \
```
Hmm, this string is always constructed where the torch check fails or passes... Reformulate to c10::LazyStr or atleast fmtlib.

kwen2501 committed:
Update on "[SymmMem] Move code to where it is used"


`maybe_initialize_env_vars` and `initialize_nvshmem_with_store` are only used in `NVSHMEMSymmetricMemory.cu`. Moving them there.

cc H-Huang awgu wanchaol fegin fduwjj wz337 wconstab d4l3k

[ghstack-poisoned]

kwen2501 commented:
@pytorchbot merge -f "CI passed; minor format change"
